### PR TITLE
feat: history page

### PR DESCRIPTION
**Pull request by sakshi006**
Added  " watch history page"

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,8 +24,9 @@ function App() {
         <Route path="/signup" element={<Signup />} />
         <Route path="/liked" element={<LikedVideo />} />
         <Route path="/watchlater" element={<WatchLater />} />
+        <Route path="/history" element={<History />} />
         <Route path="*" element={<NotFound />} />
```
Comment (Chiragmodi01) [inline src/App.js:28]: This route for 404 page is not working

File: src/Components/HistoryCard/HistoryCard.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            <img className="card-image-one" src={item.thumbnail} alt="card" />
+          </div>
+        </Link>
+        <Link className="single-video-link" to={`/video/${item._id}`}>
+          <article className="card-text-one">
+            <figcaption>{item.title}</figcaption>
+            <h5>{item.creator}</h5>
+            <summary>Published on : {item.published}</summary>
+          </article>
```
Comment (Chiragmodi01) [inline src/Components/HistoryCard/HistoryCard.jsx:24]: Always a good practice to use semantic html tags. So, it looks good here.

File: src/Pages/SingleVideo/SingleVideo.jsx [hunk 3]
```diff
@@ -3,27 +3,28 @@ import { BsFillClockFill } from "react-icons/bs";
...
-import { useLike } from "../../context/likedVideoContext";
+import {useVideoContext,useWatchLater,useLike,useHistory} from "../../context"
 
 const SingleVideo = () => {
-  const { videoID } = useParams();
-  const { video } = useVideoContext();
   const token = localStorage.getItem("token");
   const navigate = useNavigate();
+
+  // context imports
```
Comment (Chiragmodi01) [inline src/Pages/SingleVideo/SingleVideo.jsx]: Seems obvious, comment can be removed.

File: src/Pages/SingleVideo/SingleVideo.jsx [hunk 4]
```diff
@@ -3,27 +3,28 @@ import { BsFillClockFill } from "react-icons/bs";
...
   const { watchLater, addToWatchLater } = useWatchLater();
+  const { addToHistory } = useHistory();
 
+  const findVideo = video.find((item) => item._id === videoID);
   const findItemInLike = likeState.likes.find(
     (product) => product._id === videoID
   );
   const findItemInWatchLater = watchLater.watchLaterArray.find(
     (product) => product._id === videoID
   );
```
Comment (Chiragmodi01) [inline src/Pages/SingleVideo/SingleVideo.jsx:25]: You can store them in separate 'utils' file.

File: src/Pages/SingleVideo/SingleVideo.jsx [hunk 5]
```diff
@@ -33,6 +34,7 @@ const SingleVideo = () => {
         width="100%"
         height="540rem"
```
Comment (Chiragmodi01) [inline src/Pages/SingleVideo/SingleVideo.jsx:34]: Prefer to not use inline styling.

File: src/Pages/SingleVideo/SingleVideo.jsx [hunk 6]
```diff
@@ -33,6 +34,7 @@ const SingleVideo = () => {
...
         src={`https://www.youtube.com/embed/${findVideo._id}`}
+        onInferredClick={() => (token && addToHistory(findVideo))}
```
Comment (Chiragmodi01) [inline src/Pages/SingleVideo/SingleVideo.jsx:36]: `onInferredClick` is something new for me, have to check it out!

File: src/context/historyContext.jsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        });
+      }
+    } catch (err) {
+      console.error(err,"clearHistory");
+    }
+  };
+
+  useEffect(() => {
+    getHistoryData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Chiragmodi01) [inline src/context/historyContext.jsx:89]: remove comments if not necessary, or maintain a trash notes file in the system and delete before deploying.

**Review by Chiragmodi01 — COMMENTED** [review]
- In UI you can keep card's shadow mild and make 'remove' button looks like a button than just adding underline below it.
- If there's one video in history, '1 videos' doesn't look good while reading, you can just put 'video' or write logic.
Overall code structure looks good to me, just added few sanity comments.

**Review by Balakumar3018 — APPROVED** [review]
Good to go.
Suggestion:-
It would be great if you add shadow only mouse hovers on the card.

**Comment by sakshi006** [comment]
> * In UI you can keep card's shadow mild and make 'remove' button looks like a button than just adding underline below it.
> * If there's one video in history, '1 videos' doesn't look good while reading, you can just put 'video' or write logic.
>   Overall code structure looks good to me, just added few sanity comments.

Got it, will make the required changes

File: src/App.js (hunk 1 above)
Comment (sakshi006) [reply src/App.js:28]: It's working for me

File: src/context/historyContext.jsx (hunk 7 above)
Comment (sakshi006) [reply src/context/historyContext.jsx:89]: Got it

File: src/Pages/SingleVideo/SingleVideo.jsx (hunk 6 above)
Comment (sakshi006) [reply src/Pages/SingleVideo/SingleVideo.jsx:36]: Yeah, it was new for me as well

**Pull request merged**